### PR TITLE
Feature: table column groups

### PR DESCRIPTION
**Pull request by re1ro**
* [x] ResultTable: column groups in table header via `fieldGroup` prop

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 20cc231cf8d87db6a42e9936c45c46add2a1e7eb


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ac7d68b3ce9a0bdff253dbeba7f32a55077e611f
</p>

File: src/exampleTypes/ResultTable/index.js [hunk 1]
```diff
@@ -88,10 +88,47 @@ let ResultTable = ({
...
   return (
     <>
       <Table data-path={node.path}>
         <Thead>
+          {F.mapIndexed(
+            (columnGroupRow, i) => (
+              <Tr key={i}>
+                {F.mapIndexed(
+                  ({ groupName, colspan, HeaderCell = Th }, j) => (
+                    <HeaderCell key={j} colSpan={colspan}>
```
Comment (rudolph9) [inline src/exampleTypes/ResultTable/index.js:122]: do we need `i` an `j` to ensure a unique `key`?

**Review by rudolph9 — APPROVED** [review]
otherwise looks good.

File: src/exampleTypes/ResultTable/index.js (hunk 1 above)
Comment (stellarhoof) [reply src/exampleTypes/ResultTable/index.js:122]: `key` only needs to be unique within this array

**Review by stellarhoof — APPROVED** [review]

File: src/exampleTypes/ResultTable/index.js (hunk 1 above)
Comment (rudolph9) [reply src/exampleTypes/ResultTable/index.js:122]: Yeah your right, I missed the `Tr` the parent `F.mapIndexed` is nesting these in.

**Review by rudolph9 — APPROVED** [review]

**Pull request merged**